File: ServiceApp/src/main/java/com/example/ServiceApp/dto/ProductDto.java
package com.example.ServiceApp.dto;

import lombok.*;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
public class ProductDto {

    private Long id;
    private String name;
    private String cod;
    private Integer quantity;
    private Long producer;
    private String producerName;
}

File: service-interface/src/app/dtos/productDto.ts

export class ProductDto {
  id?: number;
  name?: string;
  cod?: string;
  quantity?: number;
  producer: number;
  producerName?: string;
  price?: number;
  priceWithVAT?: number;

  constructor(id: number, name: string, cod: string, quantity: number, producer: number, priceWithVAT:number, producerName: string) {
    this.id = id;
    this.name = name;
    this.cod = cod;
    this.quantity = quantity;
    this.producer = producer;
    this.producerName = producerName;
    this.priceWithVAT = priceWithVAT;

  }
}
